src/encoded/types/cart.py

```diff
@@ -99,47 +88,46 @@ def get_or_create_cart_by_user(context, request):
     user = request.registry[COLLECTIONS]['user'].get(userid)
     if not user:
         raise HTTPBadRequest()
-    carts = _get_carts_by_user(request, userid, ['disabled', 'deleted '])
+    carts = get_cart_objects_by_user(request, userid, blocked_statuses)
+    return carts, user
+
+
+@view_config(context=Cart.Collection, request_method='GET', permission='save-carts', name='get-cart')
+def get_or_create_cart_by_user(context, request):
+    carts, user = _get_user_carts(request, ['disabled', 'deleted '])
     if not carts:
         cart = _create_cart(request, user)
+        cart_atids = None
+    else:
+        cart_atids = [cart['@id'] for cart in carts]
     request.response.status = 200
     return {
         'status': 'success',
         '@type': ['result'],
-        '@graph': carts or [cart]
+        '@graph': cart_atids or [cart]
     }
 
 
 @view_config(context=Cart.Collection, request_method='PUT', permission='save-carts', name='put-cart')
 def create_cart_by_user(context, request):
-    userid = [
-        p.replace('userid.', '')
-        for p in request.effective_principals
-        if p.startswith('userid.')
-    ]
-    if not userid:
-        raise HTTPBadRequest()
-    else:
-        userid = userid[0]
-    user = request.registry[COLLECTIONS]['user'].get(userid)
-    if not user:
-        raise HTTPBadRequest()
-    cart_name = None
-    user_carts = get_cart_objects_by_user(request, userid)
-    visible_user_carts = filter_carts_to_visible(request, user_carts)
-    countable_carts = [
-        cart
-        for cart in visible_user_carts
-        if cart['status'] != 'disabled'
-    ]
-    cart_atids = [cart['@id'] for cart in visible_user_carts] if visible_user_carts else []
+    user_carts, user = _get_user_carts(request, ['deleted'] if not 'group.admin' in request.effective_principals else [])
+    cart_atids = [cart['@id'] for cart in user_carts]
     cart_status = request.json.get('status', None)
     cart_name = request.json.get('name', '').strip()
     # User writing a new cart; check for cart overflow and naming conflicts
+    if cart_status != 'disabled':
+        countable_carts = [
+            cart
+            for cart in user_carts
+            if cart['status'] != 'disabled' and cart['status'] != 'deleted'
+        ]
+    else:
```

Comment:
Just set countable_carts = [] at the top of the function to avoid this else statement.

Reply:
What's the problem with else?

Reply:
It's just not really doing much here. The if statement catches everything but disabled. In my mind it makes sense to start with an empty cart and add to it based on the condition, rather than having forking paths.

Reply:
I made this change, but for the record, I don’t like that I’m assigning to countable_carts potentially twice when before I was always assigning it once.